v4l2.go

```diff
@@ -418,28 +457,52 @@ func waitForFrame(fd uintptr, timeout uint32) (count int, err error) {
 
 }
 
+func getControl(fd uintptr, id uint32) (int32, error) {
+	ctrl := &v4l2_control{}
+	ctrl.id = id
+	err := ioctl.Ioctl(fd, VIDIOC_G_CTRL, uintptr(unsafe.Pointer(ctrl)))
+	return ctrl.value, err
+}
+
 func setControl(fd uintptr, id uint32, val int32) error {
 	ctrl := &v4l2_control{}
 	ctrl.id = id
 	ctrl.value = val
 	return ioctl.Ioctl(fd, VIDIOC_S_CTRL, uintptr(unsafe.Pointer(ctrl)))
 }
 
-func getControls(fd uintptr) map[uint32]string {
-	query := &v4l2_queryctrl{}
-	var controls map[uint32]string
+func queryControls(fd uintptr) map[string]control {
```

Comment:
this function should return ```([]control, error)```
We shouldn't assemble map here, only list. Map can be assembled in Webcam::GetControlList()

Reply:
OK